<launch>
  <arg name="robot_name"                        default=""/>
  <arg name="robot_model"                       default="$(arg robot_name)"/>
  <arg name="port"                              default="/dev/ttyDXL"/>
  <arg name="turret_operating_mode"             default="position"/>
  <arg name="turret_profile_velocity"           default="0"/>
  <arg name="turret_profile_acceleration"       default="0"/>
  <arg name="use_time_based_profile"            default="false"/>
  <arg name="use_world_frame"                   default="true"/>
  <arg name="external_urdf_loc"                 default=""/>
  <arg name="use_default_rviz"                  default="false"/>

  <include file="$(find interbotix_descriptions)/launch/description.launch">
    <arg name="robot_name"                      value="$(arg robot_name)"/>
    <arg name="robot_model"                     value="$(arg robot_model)"/>
    <arg name="use_default_gripper_bar"         value="false"/>
    <arg name="use_default_gripper_fingers"     value="false"/>
    <arg name="use_world_frame"                 value="$(arg use_world_frame)"/>
    <arg name="external_urdf_loc"               value="$(arg external_urdf_loc)"/>
    <arg name="use_default_rviz"                value="$(arg use_default_rviz)"/>
  </include>

  <node
    name="arm_node"
    pkg="interbotix_sdk"
    type="arm_node"
    output="screen"
    ns="$(arg robot_name)">
    <param name="robot_name"                    value="$(arg robot_name)"/>
    <param name="robot_model"                   value="$(arg robot_model)"/>
    <param name="motor_configs"                 value="$(find interbotix_sdk)/config/"/>
    <param name="port"                          value="$(arg port)"/>
    <param name="default_gripper_bar"           value="false"/>
    <param name="default_gripper_fingers"       value="false"/>
    <param name="arm_operating_mode"            value="$(arg turret_operating_mode)"/>
    <param name="arm_profile_velocity"          value="$(arg turret_profile_velocity)"/>
    <param name="arm_profile_acceleration"      value="$(arg turret_profile_acceleration)"/>
    <param name="gripper_operating_mode"        value="none"/>
    <param name="gripper_profile_velocity"      value="0"/>
    <param name="gripper_profile_acceleration"  value="0"/>
    <param name="use_moveit"                    value="false"/>
    <param name="use_time_based_profile"        value="$(arg use_time_based_profile)"/>
  </node>

</launch>
